### PR TITLE
Refactored cookiecutter variables, overhauled docs

### PR DESCRIPTION
**Pull request by ewels**
So this accidentally ended up being quite a lot of changes. And I didn't even break them up into different commits. Sorry...

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ install:
 script:
   - python -m pytest --cov=nf_core .
   - nf-core create -n testpipeline -d "This pipeline is for testing"
-  - nf-core lint testpipeline
+  - nf-core lint nf-core-testpipeline
```
Comment (sven1103) [inline .travis.yml]: mustn't that be equal to the pipeline name you just created one line before?

Comment (sven1103) [reply .travis.yml]: ah, the create command does this automatically, nice one

File: nf_core/pipeline-template/cookiecutter.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
-    "pipeline_name": "example",
-    "pipeline_short_description": "This pipeline takes some data and does something with it.",
-    "pipeline_slug": "{{ cookiecutter.pipeline_name.lower().replace(' ', '-').replace('/', '-') }}",
+    "name": "example",
```
Comment (sven1103) [inline nf_core/pipeline-template/cookiecutter.json]: nf-core/example?

Comment (ewels) [reply nf_core/pipeline-template/cookiecutter.json]: Yup 👍

**Pull request merged**